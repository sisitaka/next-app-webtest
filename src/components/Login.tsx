const Login = () => {
  return (
    <div>
      <span>email: </span>
      <input type="text" className="border border-black" />
      <br />
      <span>password: </span>
      <input type="text" className="border border-black" />
    </div>
  );
};

export default Login;
