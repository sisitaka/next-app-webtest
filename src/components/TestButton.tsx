"use client"

import { useState } from "react";

export const TestButton = () => {
    const [state, setState] = useState(false)
    const handleClick = () => {
        setState((prevState) => !prevState)
    };
    return (
        <>
        <button placeholder="テスト" onClick={handleClick} className="p-3 rounded-lg bg-red-600 text-white">{state ? "ON" : "OFF"}</button>
        </>
    )
}
