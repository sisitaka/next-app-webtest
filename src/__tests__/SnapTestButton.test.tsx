import { TestButton } from '@/components/TestButton';
import { render } from '@testing-library/react';
test('描画されてすぐはOFFと表示されている', () => {
  const view = render(<TestButton />);
  expect(view.container).toMatchSnapshot();
});
