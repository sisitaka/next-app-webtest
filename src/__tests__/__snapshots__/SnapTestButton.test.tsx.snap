// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`描画されてすぐはOFFと表示されている 1`] = `
<div>
  <button
    class="p-3 rounded-lg bg-red-600 text-white"
    placeholder="テスト"
  >
    OFF
  </button>
</div>
`;
